diff --git a/components/sync/driver/sync_user_settings_impl.cc b/components/sync/driver/sync_user_settings_impl.cc
index 64962d18741a49c3a614d0acdd9beca8cd77df69..c537074cb35195779b29b97ea3390db4ea70cb3d 100644
--- a/components/sync/driver/sync_user_settings_impl.cc
+++ b/components/sync/driver/sync_user_settings_impl.cc
@@ -5,7 +5,8 @@
 #include "components/sync/driver/sync_user_settings_impl.h"
 
 #include <utility>
-
+    #include "base/debug/stack_trace.h"
+    #include "base/debug/task_trace.h"
 #include "base/feature_list.h"
 #include "base/logging.h"
 #include "base/metrics/histogram_macros.h"
@@ -71,10 +72,18 @@ SyncUserSettingsImpl::SyncUserSettingsImpl(
 SyncUserSettingsImpl::~SyncUserSettingsImpl() = default;
 
 bool SyncUserSettingsImpl::IsSyncRequested() const {
-  return prefs_->IsSyncRequested();
+//  return prefs_->IsSyncRequested();
+bool bret = prefs_->IsSyncRequested();
+LOG(ERROR) << "[BraveSync] " << __func__ << " 000 RET bret=" <<bret;
+return bret;
 }
 
 void SyncUserSettingsImpl::SetSyncRequested(bool requested) {
+LOG(ERROR) << "[BraveSync] " << __func__ << " 000 SET requested=" << requested;
+LOG(ERROR) << "[BraveSync] " << __func__ << " stack=" << base::debug::StackTrace().ToString();
+LOG(ERROR) << "[BraveSync] " << __func__ << " task=";
+base::debug::TaskTrace().Print();
+
   prefs_->SetSyncRequested(requested);
 }
 
